libsplinter/src/admin/service/mod.rs

```diff
@@ -553,6 +634,36 @@ mod tests {
             .propose_circuit(payload, "test".to_string())
             .expect("The proposal was not handled correctly");
 
+        // wait up to 1 second for the service protocol version request
+        let message;
+        let start = Instant::now();
+        loop {
+            if Instant::now().duration_since(start) > Duration::from_secs(1) {
```

Comment:
1 second might be too short of a duration on very bad networks.

Reply:
Since this is in a test, I would suggest 60 seconds, as that's where cargo starts to complain about a test taking too long anyway.

Reply:
Updated the test to wait for 60 seconds